### PR TITLE
[RHELC-1470] Add a test using a proxy server

### PR DESCRIPTION
**Pull request by kokesak**
Add a test using a proxy server.

Jira Issues: [RHELC-1470](https://issues.redhat.com/browse/RHELC-1470)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by kokesak** [comment]
OL7 passed: http://artifacts.osci.redhat.com/testing-farm/2b844e01-87ab-41ee-ad54-1e0e00899fb7/

File: tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                f"firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 0 -p tcp -m tcp -d {ip_address} --dport=443 -j ACCEPT"
+            ).returncode
+            == 0
+        )
+        assert (
+            shell(
+                f"firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 0 -p tcp -m tcp -d {ip_address} --dport=80 -j ACCEPT"
+            ).returncode
+            == 0
+        )
```
Comment (r0x0d) [inline tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py:44]: We can't pass --dport twice in the same command? That's a shame :(

File: tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    # Allow connection to the proxy server
+    assert (
+        shell(
+            f"firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 0 -p tcp -m tcp --dport=3128 -j ACCEPT"
```
Comment (r0x0d) [inline tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py:48]:
```suggestion
            "firewall-cmd --permanent --direct --add-rule ipv4 filter OUTPUT 0 -p tcp -m tcp --dport=3128 -j ACCEPT"
```

File: tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py (hunk 1 above)
Comment (kokesak) [reply tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py:45]: Yeah, I tried that, but it did not work.. /:

File: tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py (hunk 2 above)
Comment (kokesak) [reply tests/integration/tier1/destructive/proxy-conversion/test_proxy_conversion.py]: Thanks, fixed!

**Review by danmyway — APPROVED** [review]
Great, thanks!

**Pull request merged**